packages/yarnpkg-fslib/sources/path.ts

```diff
@@ -1,14 +1,20 @@
 import path from 'path';
 
-export type PortablePath = string & { _portable_path: true };
-export type NativePath = string & { _portable_path?: false };
+enum PathType {
```

Comment:
Note: I'd love to have used a `const enum` here because this enum is only present because we need its type, never for its value. Const enums don't have a runtime footprint while regular enums do, which is incidentally why babel doesn't support const enums at all.